sysl2/sysl/utils/strSet_test.go

```diff
@@ -0,0 +1,162 @@
+package utils
+
+import (
+	"testing"
+
+	"github.com/anz-bank/sysl/src/proto"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMakeStrSet(t *testing.T) {
+	a := MakeStrSet("a", "b", "c", "e")
+	assert.Equal(t, 4, len(a), "Unexpect result")
+}
+
+func TestMakeStrSetWithDuplicateInitialValues(t *testing.T) {
+	a := MakeStrSet("a", "b", "c", "e", "a", "a", "c")
+	assert.Equal(t, 4, len(a), "Unexpect result")
+}
+
+func TestMakeStrSetWithEmptyStringInitialValues(t *testing.T) {
+	a := MakeStrSet("a", "b", "c", "e", "a", "a", "c", "", "")
+	assert.Equal(t, 5, len(a), "Unexpect result")
+}
+
+func TestMakeStrSetWithoutInitialValues(t *testing.T) {
+	a := MakeStrSet()
+	assert.Equal(t, 0, len(a), "Unexpect result")
+}
+
+func TestMakeStrSetFromSpecificAttrWithEmptyAttrs(t *testing.T) {
+	attrs := map[string]*sysl.Attribute{}
+
+	a := MakeStrSetFromSpecificAttr("patterns", attrs)
+	assert.Equal(t, 0, len(a), "Unexpect result")
+}
+
+func TestMakeStrSetFromSpecificAttrWithoutPatternAttr(t *testing.T) {
+	attrs := map[string]*sysl.Attribute{
+		"test": {Attribute: &sysl.Attribute_S{S: "test"}},
+	}
+
+	a := MakeStrSetFromSpecificAttr("patterns", attrs)
+	assert.Equal(t, 0, len(a), "Unexpect result")
+}
+
+func TestMakeStrSetFromPatternsAttr(t *testing.T) {
+	attrs := map[string]*sysl.Attribute{
+		"patterns": {
+			Attribute: &sysl.Attribute_A{
+				A: &sysl.Attribute_Array{
+					Elt: []*sysl.Attribute{
+						{Attribute: &sysl.Attribute_S{S: "test"}},
+					},
+				},
+			},
+		},
+	}
+
+	a := MakeStrSetFromSpecificAttr("patterns", attrs)
+	assert.Equal(t, 1, len(a), "Unexpect result")
+}
+
+func TestContains(t *testing.T) {
+	a := MakeStrSet("a", "b", "c", "e")
+	assert.Equal(t, 4, len(a), "Unexpect result")
```

Comment:
"Unexpect result" => "Unexpected result"

Reply:
Replaced all the wording error.